renderdoc/driver/vulkan/wrappers/vk_descriptor_funcs.cpp

```diff
@@ -934,7 +962,17 @@ void WrappedVulkan::ReplayDescriptorSetWrite(VkDevice device, const VkWriteDescr
             }
           }
 
-          (*bind)[curIdx].SetBuffer(writeDesc.descriptorType, writeDesc.pBufferInfo[d]);
+          if(writeDesc.descriptorType == VK_DESCRIPTOR_TYPE_ACCELERATION_STRUCTURE_KHR)
+          {
+            VkWriteDescriptorSetAccelerationStructureKHR *asDesc =
+                (VkWriteDescriptorSetAccelerationStructureKHR *)(writeDesc.pNext);
```

Comment:
Can't assume `pNext`'s type here. This would be a good candidate to cache outside of the loop as well (although I don't expect anyone will have huge arrays of ASs, it's equally clear and more optimal).